nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/deck_guns.dm

```diff
@@ -267,16 +286,107 @@
+	var/obj/machinery/deck_turret/payload_gate/payload_gate
+	///The linked deck turret computer.
+	var/obj/machinery/computer/deckgun/computer
+
+/obj/machinery/deck_turret/core/Destroy(force)
+	unlink_parts()
+	return ..()
+
+/obj/machinery/deck_turret/core/multitool_act(mob/living/user, obj/item/I)
+	..()
+	update_parts()
+	var/list/specialbits = list()
+	if(!anchored)
+		specialbits += "NOT ANCHORED"
+	if(!turret)
+		specialbits += "NO TURRET FOUND"
+	if(!payload_gate)
+		specialbits += "NO PAYLOAD GATE"
+	if(!length(powder_gates))
+		specialbits += "NO POWDER GATES"
+	to_chat(user, "<span class='notice'>Linkages updated.</span>")
+	if(length(specialbits))
+		to_chat(user, "<span class='warning'>ERRORS DETECTED: [jointext(specialbits, ", ")]</span>")
+	return TRUE
+
+/obj/machinery/deck_turret/core/default_unfasten_wrench(mob/user, obj/item/I, time)
+	. = ..()
+	if(. != SUCCESSFUL_UNFASTEN)
+		return
+	unlink_parts()
+
+
+/**
+ * Unlinks linked machines from the deckgun core.
+ * * also_unlink_turret: When true, also unlinks the deckgun turret itself.
+ */
+/obj/machinery/deck_turret/core/proc/unlink_parts(also_unlink_turret = TRUE)
+	payload_gate?.core = null
+	payload_gate = null
+	for(var/obj/machinery/deck_turret/powder_gate/powder_gate in powder_gates)
+		powder_gate.core = null
 	powder_gates = list()
-	computer = locate(/obj/machinery/computer/deckgun) in orange(1, src)
-	computer.core = src
+	computer?.core = null
+	computer = null
+	if(also_unlink_turret)
+		turret?.core = null
+		turret = null
+
+/**
+ * Recalculates linkage of the deck turret core's parts. Unlinks all parts except the turret beforehand.
+ * Only detects one machine per turf checked, and ignores the core's turf.
+ * Does not link the turret itself.
+ */
+/obj/machinery/deck_turret/core/proc/update_parts()
+	if(!anchored)
+		return
+	unlink_parts(FALSE)
+	//Build linkage by turf. Only one machine is aknowledged per turf.
+	for(var/turf/T in (RANGE_TURFS(1, src) - get_turf(src)))
+		for(var/obj/machinery/maybe_machine in T)
+			if(istype(maybe_machine, /obj/machinery/deck_turret/powder_gate))
+				var/obj/machinery/deck_turret/powder_gate/maybe_powder_gate = maybe_machine
+				if(maybe_powder_gate.core)
+					continue
+				if(!maybe_powder_gate.anchored)
+					continue
+				powder_gates += maybe_powder_gate
+				maybe_powder_gate.core = src
+				break
+			if(!computer && istype(maybe_machine, /obj/machinery/computer/deckgun))
+				var/obj/machinery/computer/deckgun/guncomputer = maybe_machine
+				if(guncomputer.core)
+					continue
+				computer = guncomputer
+				guncomputer.core = src
+				break
+			if(!payload_gate && istype(maybe_machine, /obj/machinery/deck_turret/payload_gate))
+				var/obj/machinery/deck_turret/payload_gate/maybe_payload_gate = maybe_machine
+				if(maybe_payload_gate.core)
+					continue
+				if(!maybe_payload_gate.anchored)
+					continue
+				payload_gate = maybe_payload_gate
+				maybe_payload_gate.core = src
+				break
```

Comment:
```suggestion
		for(var/obj/machinery/maybe_machine in T)
			switch(maybe_machine.type)
				if(/obj/machinery/deck_turret/powder_gate)
					var/obj/machinery/deck_turret/powder_gate/maybe_powder_gate = maybe_machine
					if(maybe_powder_gate.core)
						continue
					if(!maybe_powder_gate.anchored)
						continue
					powder_gates += maybe_powder_gate
					maybe_powder_gate.core = src
					break
				if(/obj/machinery/computer/deckgun)
					if(computer)
						continue
					var/obj/machinery/computer/deckgun/guncomputer = maybe_machine
					if(guncomputer.core)
						continue
					computer = guncomputer
					guncomputer.core = src
					break
				if(/obj/machinery/deck_turret/payload_gate)
					if(payload_gate)
						continue
					var/obj/machinery/deck_turret/payload_gate/maybe_payload_gate = maybe_machine
					if(maybe_payload_gate.core)
						continue
					if(!maybe_payload_gate.anchored)
						continue
					payload_gate = maybe_payload_gate
					maybe_payload_gate.core = src
					break
```
Could use a switch but I guess you do lose out on checking the gate/computer before the type in that.